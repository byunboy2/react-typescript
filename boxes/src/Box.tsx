interface BoxInterface {
  id: string;
  width: number;
  height: number;
  backgroundColor: string;
  remove: Function;
}

/** Colored box presentation, along with removeButton
 *
 * Props:
 * - id (unique)
 * - width
 * - height
 * - backgroundColor
 * - remove (function to call)
 *
 * BoxList -> Box
 */

const Box: React.FC<BoxInterface> = ({ id, width = 5, height = 5, backgroundColor, remove }) => {
  /** Remove a box. */
  function handleRemove() {
    remove(id);
  }

  return (
    <div className="Box">
      <div
        className="Box-box"
        data-testid={`box-${id}`}
        style={{
          height: `${height}em`,
          width: `${width}em`,
          backgroundColor: backgroundColor,
        }}
      />
      <button className="Box-removeBtn" onClick={handleRemove}>
        Remove The Box!
      </button>
    </div>
  );
}

export default Box;
