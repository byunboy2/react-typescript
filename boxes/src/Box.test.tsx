import { render } from "@testing-library/react";
import Box from "./Box";

describe("Box component", () => {
  it("renders without crashing", function () {
    render(
      <Box
        id={"1"}
        width={10}
        height={10}
        backgroundColor={"red"}
        remove={() => {}}
      />
    );
  });

  it("matches the snapshot", function () {
    const { container } = render(
      <Box
        id={"1"}
        width={10}
        height={10}
        backgroundColor={"red"}
        remove={() => {}}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("renders with correct styles", function () {
    const { getByTestId } = render(
      <Box
        id={"1"}
        width={10}
        height={10}
        backgroundColor={"red"}
        remove={() => {}}
      />
    );

    const box = getByTestId("box-1");

    expect(box).toHaveStyle(`width: 10em`);
    expect(box).toHaveStyle(`height: 10em`);
    expect(box).toHaveStyle(`background-color: red`);
  });
});
