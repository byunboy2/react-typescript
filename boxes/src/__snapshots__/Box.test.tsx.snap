// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box component matches the snapshot 1`] = `
<div>
  <div
    class="Box"
  >
    <div
      class="Box-box"
      data-testid="box-1"
      style="height: 10em; width: 10em; background-color: red;"
    />
    <button
      class="Box-removeBtn"
    >
      Remove The Box!
    </button>
  </div>
</div>
`;
